daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs

```diff
@@ -660,19 +664,22 @@ execMigrate opts inFile1 inFile2 mbDir = do
             createDirectoryIfMissing True $ takeDirectory path
             writeFile path mod
   where
-    parse loggerH fp = do
-        errOrMod <-
-            runGhcFast $
-            runExceptT $ do
-                lift $ setupDamlGHC [] Nothing []
-                -- parse without any preprocessing, so that we can see which data definitions have
-                -- already generic instances.
-                parseFileContents ((,) []) fp Nothing
-        case errOrMod of
-            Left err -> ioError $ userError $ show err
-            Right (ds, mod) -> do
-                unless (null ds) $ Logger.logWarning loggerH (T.pack $ show ds)
-                return mod
+    getEntry fp dar =
+        maybe (ioError $ userError $ "Package does not contain " <> fp) pure $
+        findEntryByPath fp dar
+    getModule modName pkg =
+        maybe
+            (ioError $
+             userError $
+             T.unpack $ "Can't find module" <> LF.moduleNameString modName)
+            pure $
+        NM.lookup modName $ LF.packageModules pkg
+    generatePkgMap logH dar =
+        withSystemTempDirectory "generatePkgMap" $ \tmpDir -> do
```

Comment:
I'd just use withTempDir from extra and avoid the dependency.

Reply:
cool, will change.